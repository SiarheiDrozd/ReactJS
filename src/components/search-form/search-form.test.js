import { fireEvent, render } from '@testing-library/react';
import SearchForm from './search-form';
import { configure } from '@testing-library/react';
import { userEvent } from '@storybook/testing-library';
import '@testing-library/jest-dom';

configure({ testIdAttribute: 'id' });
describe('1. Search-form component', () => {
  beforeEach(() => {});
  test('1.1 should render initial value', () => {
    const element = render(<SearchForm initialQuery={'dummy text'} />);
    const input = element.getByRole('textbox');

    expect(input).toHaveValue('dummy text');
  });
  test('1.2 should trigger search on button click', () => {
    const onChange = jest.fn();
    const element = render(<SearchForm onSearch={onChange} />);
    const input = element.getByRole('textbox');
    const searchButton = element.getByRole('button');

    fireEvent.change(input, { target: { value: 'search query' } });
    fireEvent.click(searchButton);

    expect(input).toHaveValue('search query');
    expect(onChange).toHaveBeenCalledWith('search query');
  });
  test('1.3 should trigger search on submit action', () => {
    const onChange = jest.fn();
    const element = render(<SearchForm onSearch={onChange} />);
    const input = element.getByRole('textbox');

    userEvent.type(input, 'search query{enter}');

    expect(input).toHaveValue('search query');
    expect(onChange).toHaveBeenCalledWith('search query');
  });
});
