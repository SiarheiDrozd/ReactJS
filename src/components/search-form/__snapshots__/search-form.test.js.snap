// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-form component should render initial value 1`] = `
<form
  className="search-form"
  name="searchForm"
  onSubmit={[Function]}
>
  <div
    className="input"
  >
    <input
      className="input__input"
      id="9e1e0721-a14d-4346-f488-3185ab8b5555"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      type="text"
      value="dummy text"
    />
  </div>
  <button
    className="button button__primary"
    type="submit"
  >
    Search
  </button>
</form>
`;
